### PR TITLE
Make `IssueHandler::delete_label` return `()`

### LINKED ISSUE #456: `IssueHandle::delete_label` panics
(Code is using `octocrab = "0.30.1"`)
When calling `IssueHandle::delete_label`, `IssueHandle` panics. This appears to be because it expects a response in Json format, but according to [github's documentation](https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#delete-a-label), the endpoint does not return any data (on success, it yields 204 No Content).

### PR DESCRIPTION
**Pull request by chorman0773**
Note: This may require a major version increase.
Fixes #456

### CODE REVIEW

File: src/api/issues.rs [hunk 1]
```diff
@@ -472,15 +472,17 @@ impl<'octo> IssueHandler<'octo> {
...
+    pub async fn delete_label(&self, name: impl AsRef<str>) -> Result<()> {
         let route = format!(
             "/repos/{owner}/{repo}/labels/{name}",
             owner = self.owner,
             repo = self.repo,
             name = name.as_ref(),
         );
 
-        self.crab.delete(route, None::<&()>).await
+        let _: crate::from_response::NoBody = self.crab.delete(route, None::<&()>).await?;
```
Comment (XAMPPRocky) [inline src/api/issues.rs:483]: You don't need that, you can just write the following

```rust
crate::map_github_error(self.crab._delete(uri, None::<&()>).await?)
    .await
    .map(drop)
```

**Comment by XAMPPRocky** [comment]
@chorman0773 Friendly ping 🙂

**Comment by manchicken** [comment]
It looks like this was fixed by #503.

**Comment by XAMPPRocky** [comment]
Thank you for your PR!

**Pull request closed without merging**